src/foam/nanos/boot/Boot.java

```diff
@@ -13,23 +13,26 @@
 import foam.dao.java.JDAO;
 import foam.dao.ProxyDAO;
 import foam.nanos.auth.Group;
-import foam.nanos.auth.Permission;
 import foam.nanos.auth.User;
 import foam.nanos.logger.Logger;
 import foam.nanos.logger.ProxyLogger;
 import foam.nanos.logger.StdoutLogger;
 import foam.nanos.script.Script;
 import foam.nanos.session.Session;
-import java.util.ArrayList;
-import java.util.List;
+
+import java.util.*;
+
 import static foam.mlang.MLang.EQ;
 
 public class Boot {
   // Context key used to store the top-level root context in the context.
   public final static String ROOT = "_ROOT_";
+  public final static String RELOADED_SERVICES = "_RELOADED_SERVICES_";
```

Comment:
Have Nanorouter listen to changes in the NSpecDAO and get rid of this.

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/2423/commits/e5fed73a234dbc4a10fdf6540fb95a181405f07a.